plugins/modules/postgresql_tablespace.py

```diff
@@ -437,8 +437,8 @@ def main():
 
     # Change autocommit to False if check_mode:
     if module.check_mode:
-        if PSYCOPG2_VERSION >= '2.4.2':
-            db_connection.set_session(autocommit=False)
+        if PSYCOPG_VERSION >= LooseVersion("2.4.2"):
+            db_connection.autocommit = False
```

Comment:
```suggestion
        if PSYCOPG_VERSION >= LooseVersion("2.4.2"):
            db_connection.autocommit = False
```
@elpavel as far as i can see `connect_to_db`  now cares about all that stuff. Should we maybe change the line a bit above:
```
db_connection, dummy = connect_to_db(module, conn_params, autocommit=True)
```
to something like:
```
db_connection, dummy = connect_to_db(module, conn_params, autocommit=True if not module.check_mode else False)
```
I'm not sure if this exact syntax works though, but the idea should be clear

Reply:
ah, i see now that it repeats:) so maybe also a separate function just to reduce code repetition?

Reply:
I'm not insisting. If you think that now is better, please just mark this as solved

Reply:
Separate function would make sense, as we use the same logic three times. The function could be moved to postgres.py, to keep the psycopg logic there.

Reply:
would you like to do it here or later (which would be also fine) ?

Reply:
I will do it here.